### ISSUE TITLE
Replace custom names with LOM names

### ISSUE THREAD

**Issue by markusschloesser** [markusschloesser/MackieC4_P3#15]
1. There are many functions defined which sound like natural language, like devices_on_tracks etc. I suggest to rename those to LOM names.
2. There are also quite a few abbreviations (tdp count etc), which obviously shorten things, but with the disadvantage of me cussing all the time, cos I can't remember what things stood for. So I suggest also replacing those with LOM names (track.device.parameter or similar)
Advantage, besides readability, is a way of more LOM - thinking.
What do you think?

**Comment by BilldarBagdar** [comment]
One of the beauties of source code repositories is "code change" history gets preserved by default, and that base layer has already been laid down.  So I think in both cases, "refactor" whatever you feel makes the code more understandable, particularly simply renaming things.  Be careful though about reusing "LOM names".  For example, this is a pointer into the "big list db" `self.t_d_bank_count[self.t_current]`.  If you rename things so it "looks like" a LOM pointer, but it still only points into the "big list db" that could be even more confusing.  and that is just to say don't be shy, be bold.  It shouldn't just "look like" LOM stuff by name, it should name LOM stuff because it is using LOM stuff.

With that said, my only request would be commit "functional changes" separately from "refactoring changes", maybe even dedicate branches to those kinds of efforts, one after the other.  Like one day you are "renaming things" so you have a branch, and the next day you start a new branch from where you left off because you will be "changing behavior".  At the end of day two you can decide to push your current branch to origin with all the updates from both days, or you could decide to push only the branch from day one because day two didn't (yet?) work out the way you were thinking, or you could do many other things, but I hope you get my point. Even a big functional change can get lost in the "sea of Diff" if it is in the same commit with a simple rename of two "class level" things like this `self.track_device_bank_max[self.track_current]`. (I imagine a change like that could account for hundreds of lines of diff)  The only reason I didn't refactor those abbreviations was because such a renaming effort seemed out of scope to me last spring.  I wanted to preserve the original as much as I could and still make heads and tails of it.

As I'm rereading that, I'm reminded of how bad I am at following the advice I just gave in my work practices.  We do code reviews, and lots of testing of course, but nobody is really scrutinizing my "commit clarity and quality" in that walled world.  I try to keep things separated, but the mice will play.

I like the idea of modeling Tracks and Devices as "stand alone classes".  So, for example, instead of two lists that keep track of the "current device bank" and "maximum device bank" for every track by track/list index, a Device class would just have that bank information available.  Every Track Class could maintain a simple List of Devices and a reference to the "selected Device" from that list.

I also like the idea of abandoning the thought of collecting information about every visible track in the entire session, and instead only focusing on interacting with the "current track" (other than track navigation.  But when we click next track, how long does it take to get all the display info from the LOM instead of "pre-populated" local lists?  All we can do is find out.)

**Comment by markusschloesser** [comment]
right now I do understand a lot of theoretical python (haven't yet fully gotten the @property thing), but using that knowledge and seeing what's what in the script is a little bit harder, so I'll mainly focus on 2. right now because it improves understandability. Plus it also takes your request into consideration about separation of functional vs refactoring changes.

Don't worry about me being bold! 😂 I already messed up the scripts more then once, trying out stuff. 

> I like the idea of modeling Tracks and Devices as "stand alone classes". So, for example, instead of two lists that keep track of the "current device bank" and "maximum device bank" for every track by track/list index, a Device class would just have that bank information available. Every Track Class could maintain a simple List of Devices and a reference to the "selected Device" from that list.

I also like the idea of abandoning the thought of collecting information about every visible track in the entire session, and instead only focusing on interacting with the "current track" (other than track navigation. But when we click next track, how long does it take to get all the display info from the LOM instead of "pre-populated" local lists? All we can do is find out.)

Two things:
1. I like the general idea as well, especially as the encodercontroller is huge. Also I don't understand why there's even clip stuff in there?? I tried ripping that out but then it got ugly fast. So focusing on Tracks and Devices first and using those for the 2 currently relevant modes is a good idea.
2. I don't know about the list, as you probably saw from the other issues, my main goal is to get rid of the "db" /indexes.
